### PR TITLE
Add provider Faceit

### PR DESCRIPTION
**Pull request by sofwar**

### CODE REVIEW

File: src/Faceit/Provider.php [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+            'grant_type' => 'authorization_code',
+        ];
+    }
+
+    protected function mapUserToObject(array $user)
+    {
+        return (new User())->setRaw($user)->map([
+            'id'       => $user['guid'],
+            'nickname' => $user['nickname'],
+            'avatar'   => isset($user['picture']) ? $user['picture'] : null,
```
Comment (atymic) [inline src/Faceit/Provider.php:74]: You can use null coalesce here (and drop php 5 support)

File: src/Faceit/composer.json [hunk 2]
```diff
@@ -0,0 +1,19 @@
+{
+    "name": "socialiteproviders/faceit",
+    "description": "Faceit OAuth2 Provider for Laravel Socialite",
+    "license": "MIT",
+    "authors": [{
+        "name": "Dzmitry Bekish",
+        "email": "ya.sofwar@yandex.com"
+    }],
+    "require": {
+        "php": "^5.6 || ^7.0",
```
Comment (atymic) [inline src/Faceit/composer.json:10]:
```suggestion
        "php": "^7.0",
```

File: src/Faceit/Provider.php (hunk 1 above)
Comment (sofwar) [reply src/Faceit/Provider.php:74]: Basically composer.json worth "5.6 | 7.0", do not want to break compatibility

Comment (sofwar) [reply src/Faceit/Provider.php:74]: > Could you please make a docs PR?
> https://github.com/SocialiteProviders/Documentation

https://github.com/SocialiteProviders/Documentation/pull/103

**Comment by atymic** [comment]
Tagged, sorry for the delay :)
https://packagist.org/packages/socialiteproviders/faceit

**Pull request merged**